packages/financial-templates-lib/src/clients/FinancialContractFactoryEventClient.js

```diff
@@ -0,0 +1,109 @@
+// A thick client for getting information about FinancialContractFactory events.
+
+class FinancialContractFactoryEventClient {
+  /**
+   * @notice Constructs new FinancialContractEventClient.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} financialContractFactoryAbi ExpiringMultiPartyCreator or PerpetualMultiPartyCreator truffle ABI object.
+   * @param {Object} web3 Web3 provider from truffle instance.
+   * @param {String} financialContractFactoryAddress Ethereum address of the factory contract deployed on the current network.
+   * @param {Integer} startingBlockNumber Offset block number to index events from.
+   * @param {Integer} endingBlockNumber Termination block number to index events until. If not defined runs to `latest`.
+   * @return None or throws an Error.
+   */
+  constructor(
+    logger,
+    financialContractFactoryAbi,
+    web3,
+    financialContractFactoryAddress,
+    startingBlockNumber = 0,
+    endingBlockNumber = null,
+    contractType = "Perpetual"
+    // Default to PerpetualMultiParty for now since the first intended user is the funding rate proposer bot
+  ) {
+    this.logger = logger;
+    this.web3 = web3;
+
+    // Factory contract
+    this.financialContractFactory = new this.web3.eth.Contract(
+      financialContractFactoryAbi,
+      financialContractFactoryAddress
+    );
+    this.financialContractFactoryAddress = financialContractFactoryAddress;
+
+    // Factory Contract Events data structure to enable synchronous retrieval of information.
+    this.createdContractEvents = [];
+
+    // First block number to begin searching for events after.
+    this.firstBlockToSearch = startingBlockNumber;
+
+    // Last block number to end the searching for events at.
+    this.lastBlockToSearchUntil = endingBlockNumber;
+    this.lastUpdateTimestamp = 0;
+
+    this.contractType = contractType;
```

Comment:
Should we check that the input is a valid type? What happens if someone passes "Perpteual"?

Reply:
Done good catch